core/src/core/block.rs

```diff
@@ -109,6 +111,12 @@ impl From<keychain::Error> for Error {
 	}
 }
 
+impl From<std::num::ParseIntError> for Error {
```

Comment:
I suspect this ties a pretty general `ParseIntError` to a very specific `HexError` impl.
We could potentially fail to "parse an int" in a multitude of different scenarios (now or in the future) and not all of them will be caused by hex serialization related problems.

It may make more sense to just explicitly convert one error to another closer to where the error actually occurs.

Reply:
oops lol yeah that's probably better for future proofing